repo/maintenance/maintenance_quick_test.go

```diff
@@ -0,0 +1,217 @@
+
+	err = maintenance.SetParams(ctx, env.RepositoryWriter, maintParams)
+	require.NoError(t, err)
+
+	require.NoError(t, env.RepositoryWriter.Flush(ctx))
+
+	// verify the owner was set
+	maintParams, err = maintenance.GetParams(ctx, env.Repository)
+	require.NoError(t, err)
+	require.Equal(t, co.UsernameAtHost(), maintParams.Owner)
+
+	// verify epoch manager is enabled
+	dr, isDirect := env.Repository.(repo.DirectRepository)
+	require.True(t, isDirect)
+	require.NotNil(t, dr)
+
+	fm := dr.FormatManager()
+	require.NotNil(t, fm)
+
+	mp, err := fm.GetMutableParameters(ctx)
+	require.NoError(t, err)
+	require.True(t, mp.EpochParameters.Enabled)
+
+	// verify quick maintenance has NOT run yet
+	sch, err := maintenance.GetSchedule(ctx, env.RepositoryWriter)
+
+	require.NoError(t, err)
+	require.True(t, sch.NextFullMaintenanceTime.IsZero(), "unexpected NextFullMaintenanceTime")
+	require.True(t, sch.NextQuickMaintenanceTime.IsZero(), "unexpected NextQuickMaintenanceTime")
+
+	err = snapshotmaintenance.Run(ctx, env.RepositoryWriter, maintenance.ModeQuick, false, maintenance.SafetyFull)
+	require.NoError(t, err)
+
+	// verify quick maintenance was run
+	sch, err = maintenance.GetSchedule(ctx, env.RepositoryWriter)
+
+	require.NoError(t, err)
+
+	require.NotEmpty(t, sch.Runs, "maintenance runs")
+	require.False(t, sch.NextQuickMaintenanceTime.IsZero(), "unexpected NextQuickMaintenanceTime")
+	require.True(t, sch.NextFullMaintenanceTime.IsZero(), "unexpected NextFullMaintenanceTime")
+
+	verifyEpochTasksRanInQuickMaintenance(t, ctx, env.RepositoryWriter)
+}
+
+func TestQuickMaintenanceAdvancesEpoch(t *testing.T) {
+	t.Parallel()
+
+	ft := faketime.NewAutoAdvance(time.Date(2024, time.October, 18, 0, 0, 0, 0, time.UTC), time.Second)
+	ctx, env := repotesting.NewEnvironment(t, format.FormatVersion3, repotesting.Options{
+		OpenOptions: func(o *repo.Options) {
+			o.TimeNowFunc = ft.NowFunc()
+		},
+	})
+
+	// set the repository owner since it is not set by NewEnvironment
+	maintParams, err := maintenance.GetParams(ctx, env.Repository)
+	require.NoError(t, err)
+
+	co := env.Repository.ClientOptions()
+	require.NotZero(t, co)
+
+	maintParams.Owner = co.UsernameAtHost()
+	maintenance.SetParams(ctx, env.RepositoryWriter, maintParams)
+
+	require.NoError(t, err)
+	require.NoError(t, env.RepositoryWriter.Flush(ctx))
+
+	maintParams, err = maintenance.GetParams(ctx, env.Repository)
+	require.NoError(t, err)
+	require.Equal(t, co.UsernameAtHost(), maintParams.Owner)
+
+	// verify epoch manager is enabled
+	dr, isDirect := env.Repository.(repo.DirectRepository)
+	require.True(t, isDirect)
+	require.NotNil(t, dr)
+
+	fm := dr.FormatManager()
+	require.NotNil(t, fm)
+
+	mp, err := fm.GetMutableParameters(ctx)
+	require.NoError(t, err)
+	require.True(t, mp.EpochParameters.Enabled, "epoch manager not enabled")
+
+	emgr, enabled, err := dr.ContentReader().EpochManager(ctx)
+	require.NoError(t, err)
+	require.True(t, enabled, "epoch manager not enabled")
```

Comment:
nit: since this is shared across tests and is generic sanity checking to ensure the test can continue it could be useful to factor it out into a separate helper function to help reduce copy/paste clutter in the future

Reply:
Done.
- #4191